doc/sphinx/source/esmvalcore/recipe.inc

```diff
@@ -1,5 +1,177 @@
+ESMValTool recipe
+*****************
+
+Recipes are the instructions telling ESMValTool
+about the user who wrote the recipe,
+the datasets which need to be run,
+the preprocessors that need to be applied,
+and the diagnostics which need to be run over the preprocessed data.
+This information is provided to ESMValTool in the recipe sections:
+`Documentation`_, `Datasets`_, `Preprocessors`_ and `Diagnostics`_, respectively.
+
+
+Documentation
+=============
+
+The documentation section includes:
+
+- The recipe's author's user name
+- A description of the recipe
+- The user name of the maintainer
+- A list of scientific references
+- the project or projects associated with the recipe.
+
+For example, please see the documentation section from the recipe:
+recipe_ocean_amoc.yml.
+
+.. code-block:: yaml
+
+    documentation:
+      description: |
+        Recipe to produce time series figures of the derived variable, the
+        Atlantic meriodinal overturning circulation (AMOC).
+        This recipe also produces transect figures of the stream functions for
+        the years 2001-2004.
+
+      authors:
+        - demo_le
+
+      maintainer:
+        - demo_le
+
+      references:
+        - demora2018gmd
+
+      projects:
+        - ukesm
+
+Note that the author, project, and references will need to be included in the
+config-references.yml file. The author name uses the format:
+last name (clipped after 4 letters) underscore first name (clipped to two
+letters) and all letters are lowercase. For instance, Mickey Mouse would be:
+mous_mi. Also note that this username is unlikely to be the same as the github
+user name.
+
+
+
+Datasets
+========
+
+The datasets section includes:
+
+- dataset name
+- Project (CMIP5 or 6, observations...)
+- experiment (historical/ RCP8.5 etc...)
+- Ensemble member
+- The time range
+- The model grid, gn or gr, (CMIP6 only).
+
+For example, a datasets section could be:
+
+.. code-block:: yaml
+
+    datasets:
+      - {dataset: CanESM2, project: CMIP5, exp: historical, ensemble: r1i1p1, start_year: 2001, end_year: 2004}
+      - {dataset: UKESM1-0-LL, project: CMIP6, exp: historical, ensemble: r1i1p1f2, start_year: 2001, end_year: 2004, grid: gn}
+
+
+Note that this section is not required, as datasets can also be provided in the
+`Diagnostics`_ section.
+
+
+Preprocessors
+=============
+
+The preprocessor section of the recipe includes one or more preprocesors, each
+of which may call one or several preprocessor functions.
+
+Each preprocessor section includes:
+
+- A preprocessor name.
+- A list of preprocesor functions to apply
+- Any Arguments given to the preprocessor functions.
+- The order that the preprocesor functions are applied can also be specified.
+
+The following preprocessor is an example of a preprocessor that contains
+multiple preprocessor functions:
+
+.. code-block:: yaml
+
+    preprocessors:
+      prep_timeseries: # For regional volume averaged (at the surface)
+        custom_order: true
```

Comment:
```suggestion
```
using a custom order is an advanced feature that should probably not be in the first example that users see

Reply:
Okay, I've changed the example to something simpler.